lib/java/os-environment/src/main/java/org/enso/os/environment/chdir/WorkingDirectory.java

```diff
@@ -2,7 +2,12 @@
 
 import org.enso.common.Platform;
 
-/** Class responsible for changing the working directory via native code. */
+/**
+ * Class responsible for changing the working directory via native code.
+ *
+ * <p>Note that it is not possible to change the working directory in pure Java. It has to be done
+ * via system calls.
+ */
 public abstract sealed class WorkingDirectory
```

Comment:
I find it a bit surprising that we seem to have a number of methods in this class that operate on `String`s rather than `Path`s. Wouldn't that create unnecessary conversion issues that could be easily avoided?

Reply:
In whole `WorkingDirectory`, we have to avoid anything from `java.io` as much as possible. Therefore, there is no `java.nio.file.Path` used.